src/product.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Product } from './product.types';
+
+export class ProductService {
+    private products: Product[] = [];
+
+    public addProduct(product: Product): void {
+        this.products.push(product);
+    }
+
+    public find(name: string): Product | undefined {
+        for (let i = 0; i < this.products.length; i++) {
+            if (this.products[i].name.toLowerCase() === name.toLowerCase()) {
```

Comment:
name.toLowerCase should be made before the for loop and stored in a new variable. This means that the calculation does not have to be repeated in every step => runtime savings